test/replication-luatest/gh_5568_read_only_reason_test.lua

```diff
@@ -0,0 +1,287 @@
+local t = require('luatest')
+local cluster = require('test.luatest_helpers.cluster')
+local helpers = require('test.luatest_helpers')
+
+--
+-- gh-5568: ER_READONLY needs to carry additional info explaining the reason
+-- why the error happened. It should help the clients to do better error logging
+-- and to use the provided info to find the actual leader if the reason is in
+-- being not a leader.
+--
+
+--
+-- Make functions which will create and destroy a cluster. Usage of the closures
+-- is a workaround for luatest hooks not accepting their group as a parameter.
+--
+local function make_create_cluster(g) return function(param)
```

Comment:
luacheck complains: param is unused

Reply:
Wooops, dropped.
<details><summary>diff</summary>

```Diff
@@ -13,7 +13,7 @@ local helpers = require('test.luatest_helpers')
 -- Make functions which will create and destroy a cluster. Usage of the closures
 -- is a workaround for luatest hooks not accepting their group as a parameter.
 --
-local function make_create_cluster(g) return function(param)
+local function make_create_cluster(g) return function()
     g.cluster = cluster:new({})
     local master_uri = helpers.instance_uri('master')
     local replica_uri = helpers.instance_uri('replica')
```
</details>